float64  x
float64  y
float64  z
float64  r
uint8     theta_offset